openproblems/tasks/spatial_decomposition/methods/nmfreg.py

```diff
@@ -0,0 +1,121 @@
+from ....tools.decorators import method
+from ....tools.utils import check_version
+from .._utils import split_sc_and_sp
+
+import numpy as np
+
+
+@method(
+    method_name="NMF-reg",
+    paper_name="Slide-seq: A scalable technology for measuring genome-wide expression at high spatial resolution",  # noqa: E501
+    paper_url="https://science.sciencemag.org/content/363/6434/1463",  # noqa: E501
+    paper_year=2019,
+    code_url="https://github.com/tudaga/NMFreg_tutorial",
+)
+def nmfreg(adata, test=False):
+    """NMF-reg: NMF regression for array-based spatial transcriptomics data.
+
+    Re-implementation from https://github.com/tudaga/NMFreg_tutorial.
+    Originally developed for Slide-seq data.
+
+    Parameters
+    ----------
+    adata : AnnData
+        Adata with true proportions and signature matrix.
+
+    Returns
+    -------
+    Adata with predicted proportions saved in `adata.obsm["proportions_pred"]`.
+    """
+    from scipy.optimize import nnls
+    from scipy.sparse import issparse
+    from sklearn.decomposition import NMF
+    from sklearn.preprocessing import StandardScaler
+
+    adata_sc, adata = split_sc_and_sp(adata)
+
+    n_types = adata_sc.obs["label"].cat.categories.shape[0]
+
+    factors = 30  # TODO(handle hyper params)
```

Comment:
Handle hyper parameters, see `label_projection.methods.scvi_tools` for best practices